pt/lessons/ecto/associations.md

```diff
@@ -0,0 +1,505 @@
+
+#### A Migração Has Many
+
+Vamos gerar uma migração para `Movie`:
+
+```console
+mix ecto.gen.migration create_movies
+```
+
+Abra o arquivo da migração recém gerada e defina a sua função `change`, com o intuito de criar a tabela `movies`:
+
+```elixir
+# priv/repo/migrations/*_create_movies.exs
+defmodule Example.Repo.Migrations.CreateMovies do
+  use Ecto.Migration
+
+  def change do
+    create table(:movies) do
+      add :title, :string
+      add :tagline, :string
+    end
+  end
+end
+```
+
+#### O Schema Has Many
+
+Nós vamos adicionar um esquema que especifica a relação "has many" entre um filme e os seus
+personagens.
+
+```elixir
+# lib/example/movie.ex
+defmodule Example.Movie do
+  use Ecto.Schema
+
+  schema "movies" do
+    field :title, :string
+    field :tagline, :string
+    has_many :characters, Example.Character
+  end
+end
+```
+
+A macro `has_many/3` não adiciona dados ao banco de dados por sí só. O que ela faz é utilizar uma chave estrangeira no esquema associado (`characters`)
+para tornar as associações de personagens de um filme disponíveis. Isso é o que nos permite realizar chamadas como `movie.characters`.
+
+#### A migração Belongs
+
+Agora nós estamos prontos para construir nossa migração e `schema` para `Character`. Um personagem pertence(`belongs to`) a um filme, então vamos definir uma migração que especifique o relacionamento.
+
+Primeiro, precisamos gerar a migração:
+
+```console
+mix ecto.gen.migration create_characters
+```
+
+Para declarar que um personagem pertence a um filme, precisamos da tabela `characters` e que ela possua uma coluna `movie_id`. Nós queremos que essa coluna funcione como uma chave estrangeira. Podemos alcançar isso com a seguinte linha, na chamada para `create_table/1`:
+
+```elixir
+add :movie_id, references(:movies)
+```
+
+Assim, nossa migração deve ser algo como:
+
+```elixir
+# priv/migrations/*_create_characters.exs
+defmodule Example.Repo.Migrations.CreateCharacters do
+  use Ecto.Migration
+
+  def change do
+    create_table(:characters) do
+      add :name, :string
+      add :movie_id, references(:movies)
+    end
+  end
+end
+```
+
+#### O Schema Belongs To
+
+Nosso esquema precisa definir a relação `belongs to` entre um personagem e seu filme,
+
+```elixir
+# lib/example/character.ex
+
+defmodule Example.Character do
+  use Ecto.Schema
+
+  schema "characters" do
+    field :name, :string
+    belongs_to :movie, Example.Movie
+  end
+end
+```
+
+Vamos dar uma olhada mais a fundo na macro `belongs_to/3`. Ao invés de adicionar a coluna `movie_id` na tabela `characters`, essa macro _não_ adiciona nada ao banco de dados. Ela nos _permite_ acessar os esquemas de `movies` associados através de `characters`. Ela utiliza a chave estrangeira `movie_id` para tornar os `characters` associados com um dado filme quando executamos a consulta sobre os personagens. Isso nos permite chamar `character.movie`.
```

Comment:
@lucaspbordignon WDYT about include ` faz por nós` in the end of the first phrase of this paragraph. Seems to be a more complete translation of the original phrase `Let's take a closer look at what the `belongs_to/3` macro does for us.`

```diff
- Vamos dar uma olhada mais a fundo na macro `belongs_to/3`.
+ Vamos dar uma olhada mais a fundo no que a macro `belongs_to/3` faz por nós.
```